components/list/list.js

```diff
@@ -0,0 +1,71 @@
+import './list-item.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+
+class List extends LitElement {
+	static get properties() {
+		return {
+			divider: { type: String }
+		};
+	}
+
+	static get styles() {
+		const layout = css`
+			:host {
+				display: block;
+			}
+			ul {
+				list-style-type: none;
+				padding: 0;
+			}
+			::slotted(*) {
+				display: none;
+			}
+			::slotted(d2l-list-item) {
```

Comment:
I think ::slotted(...) only targets immediate children of the slot, and would be problematic in either of those cases. Is totally an issue.

```
<d2l-list>
    <d2l-some-hm-list-item href="..." >
</d2l-list>
```
This use case won't work unless I add a mixin that is required by `d2l-some-hm-list-item` elements. In which case the mixin could apply the proper styles so `::slotted(.d2l-list-item-text-secondary-responsive)...` would still work.

```
<d2l-list>
    <d2l-list-item>
        <d2l-some-hm-component href="">
    </d2l-list-item>
</d2l-list>
```
This is okay if they are not filling the list with text and secondary text. If we are using the pattern then the HM component should do:

```
<d2l-list>
    <d2l-list-item>
        <...>Text</...>
        <...>Secondary</...>
    </d2l-list-item>
</d2l-list>
```

Reply:
Perhaps if we are still going to use `d2l-list-item-content` and `d2l-list-item-content-clickable` we can get away from the the `::slotted` selector for targeting distributed children:
```
<d2l-list>
    <d2l-list-item>
        <d2l-some-hm-component href="">
            <d2l-list-item-content></d2l-list-item-content>
        </d2l-some-hm-component>
    </d2l-list-item>
</d2l-list>
```

Reply:
`<d2l-list-item-content text="..." secondary-text="..."></d2l-list-item-content>` I actually did that and still no good. 

Also, sadly `secondary-text` being an attribute will be useless for HM implantation. Most secondary text is complex dates that tend to be in the `d2l-subtitle / d2l-list-horizontal / the better name that escape me`

Reply:
A mixin might solve this problem. I going to look into that.

Reply:
Actually, I don't think it will work.

I think we should look this over after I put in something for selecting and actions?

It would give us a better way to figure out a better solution.

Reply:
I know @dlockhart has 5 or 6 examples of what we were expecting the mark-up to look like.  I'd start with that, and layer in where the HM component will be, making sure selectors won't be broken, and everything is in the correct DOM scope in order for things like select-all to work.

Reply:
Thinking about where the HM component should be there are two cases:
`<d2l-list-item>` should be an HM component.
or
None of them should be.
With the `siren-sdk` doing these two things are a lot easier.

Getting the DOM in scope is hard with slotted elements. 

One option is to have 2 other components. 
```
<header>
<supporting-text>
```